helpers/lib/cdnify.js

```diff
@@ -26,28 +24,38 @@ module.exports = globals => {
             return '';
         }
 
-        if (/^(?:https?:)?\/\//.test(path)) {
+        if (path.startsWith('https://') || path.startsWith('http://') || path.startsWith('//')) {
             return path;
         }
-
+        // This block is to strip leading . and / from path
         if (['.', '/'].includes(path[0])) {
-            path = path.replace(/^[\.\/]+/, '');
+            for (let i = 0; i < path.length; i++) {
+                if (path[i] !== '.' && path[i] !== '/') {
+                    path = path.slice(i);
+                    break;
+                }
+            }
         }
 
-        if (protocolMatch.test(path)) {
-            var match = path.match(protocolMatch);
-            path = path.slice(match[0].length, path.length);
+        let colonIndex = path.indexOf(':');
+
+        if (colonIndex !== -1) {
+            while (colonIndex + 1 < path.length && path[colonIndex + 1] === ':') {
+                colonIndex++;
+            }
+            const protocol = path.slice(0, colonIndex + 1);
+            path = path.slice(colonIndex + 1);
```

Comment:
Tracing through this code with path = "webdav://foobar", it seems that at this point we will be left with "//foobar" and then in the next step, "/foobar", and then we will have too many "/" in the response.

Reply:
It seems like this was a bug in the old way too.  Will fix.

Reply:
fixed.